From 569d7cf2ed365480e5612f05d4082c983a28b376 Mon Sep 17 00:00:00 2001
From: Huang zhibao <hzb@rock-chips.com>
Date: Thu, 26 Nov 2020 15:36:25 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3568-nvr: set gpu and logic voltage
 fixed

NVR board logic and GPU are combined, so fixed voltage is required

Signed-off-by: Huang zhibao <hzb@rock-chips.com>
Change-Id: I5e6d52882cb11028803f037c20e0113c1259ad0c
---
 arch/arm64/boot/dts/rockchip/rk3568-nvr.dtsi | 18 ++++++++++++++++--
 1 file changed, 16 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3568-nvr.dtsi b/arch/arm64/boot/dts/rockchip/rk3568-nvr.dtsi
index e17315a4f37b..5b8da23bf996 100644
--- a/arch/arm64/boot/dts/rockchip/rk3568-nvr.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3568-nvr.dtsi
@@ -123,13 +123,23 @@
 		vin-supply = <&vcc5v0_sys>;
 	};
 
+	vdd_fixed: vdd-fixed {
+		compatible = "regulator-fixed";
+		regulator-name = "vdd_fixed";
+		regulator-min-microvolt = <900000>;
+		regulator-max-microvolt = <900000>;
+		regulator-always-on;
+		regulator-boot-on;
+		vin-supply = <&vcc5v0_sys>;
+	};
+
 	vdd_logic: vdd-logic {
 		compatible = "pwm-regulator";
 		pwms = <&pwm1 0 5000 1>;
 		regulator-name = "vdd_logic";
 		regulator-min-microvolt = <800000>;
 		regulator-max-microvolt = <1200000>;
-		regulator-init-microvolt = <950000>;
+		regulator-init-microvolt = <900000>;
 		regulator-always-on;
 		regulator-boot-on;
 		regulator-settling-time-up-us = <250>;
@@ -190,10 +200,14 @@
 };
 
 &gpu {
-	mali-supply = <&vdd_logic>;
+	mali-supply = <&vdd_fixed>;
 	status = "okay";
 };
 
+&gpu_opp_table {
+	/delete-node/ opp-800000000;
+};
+
 &hdmi {
 	status = "okay";
 };
-- 
2.35.3

